.github/workflows/main.yml

```diff
@@ -84,18 +84,20 @@ jobs:
 
   publish:
     name: Publish
-    if: (((github.event_name == 'workflow_dispatch')) && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v')) && endsWith(github.actor, '-stripe'))
-    needs: [build]
+    if: >-
+      (github.event_name == 'workflow_dispatch') &&
+      (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v')) &&
+      endsWith(github.actor, '-stripe'))
+    needs: [build, test]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
       - run: sudo apt-get install -y oathtool
       - name: Publish to NPM
         run: |
           set +x
-          echo "Preparing .npmrc"
-          echo '//registry.npmjs.org/:_authToken=$NPM_AUTH_TOKEN' > .npmrc
-          echo 'registry=https://registry.npmjs.org' >> .npmrc
+          npm config set //registry.npmjs.org/:_authToken $NPM_AUTH_TOKEN
+          npm whoami
```

Comment:
Nit: Can we add a comment for this line explaining that it's for validation and doesn't serve a critical purpose?